packages/typescript-estree/src/create-program/createProjectService.ts

```diff
@@ -22,9 +22,14 @@ export interface ProjectServiceSettings {
   service: TypeScriptProjectService;
 }
 
+export interface ProjectServiceParseSettings {
+  extraFileExtensions?: string[];
+}
+
 export function createProjectService(
   optionsRaw: boolean | ProjectServiceOptions | undefined,
   jsDocParsingMode: ts.JSDocParsingMode | undefined,
+  parseSettings?: ProjectServiceParseSettings,
```

Comment:
[Refactor] How about moving `jsDocParsingMode` into `ProjectServiceParseSettings`? They're both about parsing, after all:

```suggestion
export interface ProjectServiceParseSettings {
  extraFileExtensions?: string[];
  jsDocParsingMode?: ts.JSDocParsingMode;
}

export function createProjectService(
  optionsRaw: boolean | ProjectServiceOptions | undefined,
  parseSettings?: ProjectServiceParseSettings,
```

Reply:
I had to refactor to support an ESLint configuration that changes `extraFileExtensions` in the same linting session.  As such, I've moved the logic back to `packages/typescript-estree/src/useProgramFromProjectService.ts`.